Assets/Resources/Scripts/BreedingMenu.cs

```diff
@@ -0,0 +1,121 @@
+using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+
+public class BreedingMenu : MonoBehaviour
+{
+
+    public GameObject scrollMenuContent;
+    public GameObject droneImage;
+    public List<GameObject> allyDroneList;
+    public GameObject selectOneContent;
+    private bool selectOneSelecting;
+    public GameObject selectTwoContent;
+    private bool selectTwoSelecting;
+    public GameObject firstSelectedDrone;
+    public GameObject secondSelectedDrone;
+
+    // Start is called before the first frame update
+    void Start()
+    {
+        //Get list of living ally drones and populate scroll view with it
+        List<GameObject> droneList = GameObject.Find("Main Camera").GetComponent<Game>().livingAllies;
```

Comment:
We are accessing this gamecomponent a lot through this script. It is very ineffecient to do `.find` every time on the main camera.

Should probably load the game component into memory with a private variable so we don't have to retrieve it a bunch of times.

```
private Game gameComponent;
    void Awake()
    {
        // Cache the Game component reference
        GameObject mainCamera = GameObject.Find("Main Camera");
        if (mainCamera != null)
        {
            gameComponent= mainCamera.GetComponent<Game>();
            if (gameComponent == null)
            {
                Debug.LogError("Game component not found on Main Camera.");
            }
        }
        else
        {
            Debug.LogError("Main Camera not found.");
        }
    }
```

then the start function would look like this

```
   void Start()
    {
        if (gameComponent != null)
        {
            InitializeDroneScrollView(gameComponent.livingAllies);
        }
        ResetSelection();
    }
```

Reply:
there's a global variable for the main camera that I'm pretty sure has no overhead, that's what I use. it'd be ``Camera.main.getComponent<Game>();``. Either way it shouldn't be a huge deal if it's only called once per menu open.